networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlocksByRangeListenerWrapperTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.rpc.beaconchain.methods;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.common.primitives.UnsignedLong;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.datastructures.util.DataStructureUtil;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.networking.eth2.peers.Eth2Peer;
+import tech.pegasys.teku.networking.eth2.rpc.core.ResponseStreamListener;
+
+public class BlocksByRangeListenerWrapperTest {
+  private DataStructureUtil dataStructureUtil = new DataStructureUtil();
+  private BlocksByRangeListenerWrapper listenerWrapper;
+  private Eth2Peer peer = mock(Eth2Peer.class);
+
+  @SuppressWarnings("unchecked")
+  private ResponseStreamListener<SignedBeaconBlock> listener = mock(ResponseStreamListener.class);
+
+  @BeforeEach
+  void setUp() {
+    when(listener.onResponse(any())).thenReturn(SafeFuture.completedFuture(null));
+  }
+
+  @Test
+  void blockSlotSmallerThanFromSlot() {
+    UnsignedLong START_SLOT = UnsignedLong.valueOf(1);
+    UnsignedLong COUNT = UnsignedLong.valueOf(4);
+    UnsignedLong STEP = UnsignedLong.valueOf(2);
+    listenerWrapper = new BlocksByRangeListenerWrapper(peer, listener, START_SLOT, COUNT, STEP);
+
+    final SignedBeaconBlock block1 = dataStructureUtil.randomSignedBeaconBlock(0);
+
+    try {
+      listenerWrapper.onResponse(block1).join();
+    } catch (final Exception e) {
+      assertThat(e).isInstanceOf(BlocksByRangeResponseOutOfOrderException.class);
+    }
```

Comment:
(nit). You should be able to do something like:
```suggestion
    SafeFuture<?> result = listenerWrapper.onResponse(block1);
    assertThat(result).isCompletedExceptionally();
    assertThatThrownBy(result::get).hasCauseIntanceOf(BlocksByRangeResponseOutOfOrderException.class);
```

Reply:
Done.